synapse/handlers/receipts.py

```diff
@@ -203,20 +203,27 @@ def filter_out_private_receipts(
             for event_id, orig_event_content in room.get("content", {}).items():
                 event_content = orig_event_content
                 # If there are private read receipts, additional logic is necessary.
-                if ReceiptTypes.READ_PRIVATE in event_content:
+                if (
+                    ReceiptTypes.READ_PRIVATE in event_content
+                    or "org.matrix.msc2285.read.private" in event_content
+                ):
                     # Make a copy without private read receipts to avoid leaking
                     # other user's private read receipts..
                     event_content = {
                         receipt_type: receipt_value
                         for receipt_type, receipt_value in event_content.items()
-                        if receipt_type != ReceiptTypes.READ_PRIVATE
+                        if receipt_type
+                        not in [
+                            ReceiptTypes.READ_PRIVATE,
+                            "org.matrix.msc2285.read.private",
+                        ]
                     }
 
                     # Copy the current user's private read receipt from the
                     # original content, if it exists.
-                    user_private_read_receipt = orig_event_content[
-                        ReceiptTypes.READ_PRIVATE
-                    ].get(user_id, None)
+                    user_private_read_receipt = orig_event_content.get(
+                        ReceiptTypes.READ_PRIVATE, {}
+                    ).get(user_id, None)
```

Comment:
Does this need to care about the unstable version?

Reply:
> Or will this magically upgrade unstable ones to stable?

No, it won't.

> Does this need to care about the unstable version? 

No, I don't think so - from my understanding, we should ignore the prefixed RRs since they're unstable, we should just avoid leaking them to the client. That said, my understanding might be wrong, in that case, we should probably check for the prefixed RRs when counting notifs and also set up the migration

Reply:
Right, I guess that's OK. It *might* cause a user's read-receipt to go backwards? But doing anything else seems too complicated IMO.

Can you file a follow-up issue about cleaning up the unstable stuff at some point in the future.

Reply:
> Right, I guess that's OK. It _might_ cause a user's read-receipt to go backwards? But doing anything else seems too complicated IMO.
> 
> Can you file a follow-up issue about cleaning up the unstable stuff at some point in the future.

Actually -- if we're just going to filter them out in Python, why not do a database update that just removes them. Wouldn't that be simpler and we would need to do it eventually anyway?

Reply:
@SimonBrandner and I discussed this and depending on timing we might do a follow-up.

Reply:
Handled by https://github.com/SimonBrandner/synapse/pull/1